src/app/shared/directives/tooltip.directive.ts

```diff
@@ -0,0 +1,59 @@
+import { ApplicationRef, ComponentRef, createComponent, Directive, EventEmitter, HostListener, Input, Output, TemplateRef } from '@angular/core'
+import { TooltipComponent } from '../components/tooltip/tooltip.component'
+
+export interface TooltipMetadata {
+  xPosition: number
+  width?: number
+}
+
+@Directive({
+  selector: '[sofieTooltip]',
+})
+export class TooltipDirective {
```

Comment:
Just a reminder that we need an `OnDestroy` implemented for this to remove the tooltip if we have a `tooltipComponentRef`.

Reply:
OnDestroy has been implemented